src/libNode/FinalBlockProcessing.cpp

```diff
@@ -134,105 +126,113 @@ bool Node::LoadUnavailableMicroBlockHashes(const TxBlock& finalBlock,
+        else
         {
-            bool doRejoin = false;
-
-            if (IsMyShardMicroBlockInFinalBlock(blocknum))
+            if (shardIdsInMicroBlocks[i] == m_myshardId)
             {
-                if (m_lastMicroBlockCoSig.first != m_mediator.m_currentEpochNum)
+                if (m_microblock == nullptr)
                 {
                     LOG_GENERAL(WARNING,
-                                "Found my microblock but Cosig not updated");
+                                "Found my shard microblock but microblock obj "
+                                "not initiated");
                     doRejoin = true;
                 }
-                else
-                {
-                    toSendTxnToLookup = true;
-                }
-            }
-            else
-            {
-                if (IsMyShardIdInFinalBlock(blocknum))
+                else if (m_lastMicroBlockCoSig.first
+                         != m_mediator.m_currentEpochNum)
                 {
                     LOG_GENERAL(
                         WARNING,
-                        "Didn't found my micorblock but found shard ID");
+                        "Found my shard microblock but Cosig not updated");
                     doRejoin = true;
                 }
-            }
+                else
+                {
+                    if (m_microblock->GetHeader().GetHash()
+                            == hashesInMicroBlocks[i]
+                        && !isMicroBlockEmptys[i])
+                    {
+                        toSendTxnToLookup = true;
+                    }
+                    else
+                    {
+                        LOG_GENERAL(
+                            WARNING,
+                            "The microblock hashes in finalblock doesn't "
+                            "match with the local one"
+                                << endl
+                                << "expected: "
+                                << m_microblock->GetHeader().GetHash() << endl
+                                << "received: " << hashesInMicroBlocks[i])
+                        return false;
+                    }
+                }
 
-            if (doRejoin || m_doRejoinAtFinalBlock)
-            {
-                LOG_GENERAL(WARNING,
-                            "Failed the last microblock consensus but "
-                            "still found my shard microblock, "
-                            " need to Rejoin");
-                //RejoinAsNormal();
-                //return false;
+                break;
             }
         }
+    }
 
-        m_unavailableMicroBlocks.clear();
+    if (doRejoin || m_doRejoinAtFinalBlock)
+    {
+        // LOG_GENERAL(WARNING,
+        //             "Failed the last microblock consensus but "
+        //             "still found my shard microblock, "
+        //             " need to Rejoin");
+        // RejoinAsNormal();
+        // return false;
     }
+
```

Comment:
Since there is nothing to do inside the condition, how about comment out the condition checking, or remove them directly (if not be used in the future)?

Reply:
Good point, but still need to keep